### ISSUE TITLE
[augmentation] apply transforms to pyramid of images

### ISSUE THREAD

**Issue by bes-dev** [kornia/kornia#974]
There are some issues related to 'augmentation' module in case when we apply the same transformations to pyramid of images:
* RandomErasing generates parameters of transform related to the width/height of the image in pixels. We cannot apply these parameters to an image with another size and get the same result. It will be more clear to generate parameters in rectangle sized [1, 1].
* 'generate_parameters' doesn't implemented for RandomHorizontalFlip.

**Comment by shijianjian** [comment]
It is an interesting use case that to apply augmentations to an image pyramid. Can you elaborate more on the concrete project or use case here?

For your questions:
- I can hardly decide it here. The main concern is the reproducibility, like the crop precision might differ under different precision (e.g. float16, float32). But you have raised an interesting idea that to apply similar geometric transformations onto different sized images. I am okay to update it to change it to [0, 1] than concrete pixels, but I feel like we need to discuss more scenarios. \cc @edgarriba @ducha-aiki  Maybe having another container? Or to have anything to update the transformation matrix? 
- some operations like ```RandomHorizontalFlip``` does not need parameters at all. It is okay here.

**Comment by edgarriba** [comment]
@bes-dev I fully agree that we should support scenarios like that. Could you provide a small snippet with what you have in mind ?

**Comment by bes-dev** [comment]
@shijianjian There are some problems like image synthesis, superresolution etc. where we would like to control generation at different scales. In this way, our generator provides a multiscale image pyramid as output, and we would like to push it into a discriminator simultaneous. So, in this case, kornia should provide API that gets the image pyramid as input and returns the image pyramid (simultaneously augmented images).

What about RandomHorizontalFlip, I think that it needs a parameter like a boolean flag that says that we should or shouldn't apply flip.

@edgarriba currently I make it like this:
``` 
x - image pyramid (list of images)
transforms - nn.Sequential() of kornia transforms

for t in transforms:
    params = t.generate_parameters(...)
    for i in range(len(x)):
       x[i] = t.apply_transform(x[i], params) 
```

Currently, generate_parameters depend on input tensor, but it will be great if the transform parameters will be input-agnostic.

Also, it will be great to create more specific containers for kornia transforms (like Compose in albumentations/torchvision) that provides API like:
``` 
x - image pyramid (list of images)
transforms - nn.Sequential() of kornia transforms

params = transforms.generate_parameters(...)
for i in range(len(x)):
   x[i] = transforms.apply_transform(x[i], params) 
```

**Comment by shijianjian** [comment]
Thanks for the clarification. I like the idea of being shape-agnostic, and to decouple the parameters and input tensors.

Do you mind working on this feature? Presumably, you may modify the param_generator and apply_transform functions.

**Comment by edgarriba** [comment]
@bes-dev thanks the feedback. This is definitely a powerful feature. Let's iterate on that.
As @shijianjian suggest, if you provide a  POC we can help with the integration.

Regarding flag in `RandomHorizontalFlip`, is not `p` ? you can switch on//off by passing 0/1

**Comment by bes-dev** [comment]
@shijianjian yeah, I'll take a look at what I can do on my side.

**Comment by shijianjian** [comment]
Forgot to mention that the probability parameter is already there. It is generated automatically and you might access it by the ._param attribute.

**Comment by shijianjian** [comment]
@dkoguciuk This shall be relevant to the proposal of bounding box refactoring with floating point support #1142

**Comment by dkoguciuk** [comment]
@shijianjian , yes, thank you for letting me know :+1:

**Comment by edgarriba** [comment]
If anyone interested in this please, re-open. Thanks !
